src/templates/core/OpenAPI.hbs

```diff
@@ -1,8 +1,59 @@
+{{~#equals @root.$config.client 'angular'~}}
+import type { HttpResponse } from '@angular/common/http';
+{{~/equals~}}
+{{~#equals @root.$config.client 'axios'~}}
+import type { AxiosRequestConfig, AxiosResponse } from 'axios';
+{{~/equals~}}
+{{~#equals @root.$config.client 'node'~}}
+import type { RequestInit, Response } from 'node-fetch';
+{{~/equals~}}
+
 import type { ApiRequestOptions } from './ApiRequestOptions';
+{{#if @root.$config.useOptions}}
+{{#equals @root.$config.serviceResponse 'generics'}}
 import type { TConfig, TResult } from './types';
+{{else}}
+import type { TResult } from './types';
+{{/equals}}
+{{else}}
+import type { TResult } from './types';
+{{/if}}
 
-type Resolver<T> = (options: ApiRequestOptions) => Promise<T>;
 type Headers = Record<string, string>;
+{{#equals @root.$config.client 'angular'}}
+type Middleware<T> = (value: T) => T;
+{{else}}
+type Middleware<T> = (value: T) => T | Promise<T>;
+{{/equals}}
+type Resolver<T> = (options: ApiRequestOptions) => Promise<T>;
+
+interface Interceptor<T> {
+  _fns: ReadonlyArray<Middleware<T>>;
+  eject: (fn: Middleware<T>) => void;
+  use: (fn: Middleware<T>) => void;
+}
+
+export class Interceptors<T> {
+  _fns: Middleware<T>[];
+
+  constructor() {
+    this._fns = [];
+  }
+
+  eject(fn: Middleware<T>) {
+    const index = this._fns.indexOf(fn);
+    if (index !== -1) {
+      this._fns = [
```

Comment:
can this be `this._fns = this._fns.splice(index, 1)`

Reply:
It can, but I don't like methods that modify array in place. Much more obvious what's happening with immutable operations

Reply:
Ok, thats fine with me